pkg/query/endpointset.go

```diff
@@ -797,11 +797,7 @@ func (er *endpointRef) labelSets() []labels.Labels {
 
 	labelSet := make([]labels.Labels, 0, len(er.metadata.LabelSets))
 	for _, ls := range labelpb.ZLabelSetsToPromLabelSets(er.metadata.LabelSets...) {
-		if len(ls) == 0 {
-			continue
-		}
-		// Compatibility label for Queriers pre 0.8.1. Filter it out now.
-		if ls[0].Name == store.CompatibilityTypeLabelName {
+		if ls.Len() == 0 {
```

Comment:
This removes the ancient `store.CompatibilityTypeLabelName` which I think is okay given how old the supported version is.